test/jdk/javax/swing/JTabbedPane/TestJTabbedPaneOpaqueColor.java

```diff
@@ -0,0 +1,249 @@
+              - the area behind tabs is transparent (it must be green).
+              - the tabs area is opaque (it must be red, except the selected tab which must be gray).
+              - the content area is opaque (it must be gray).
+
+            Test Case 1 - Test TabPane Opacity:
+            NOTE: For Nimbus LAF, tabs color are always gray in color.
+
+            Verify the following with 'opaque' option:
+            when checked:
+              - the area behind tabs is opaque (it must be red).
+              - the tabs area is opaque (it must be red, except the selected tab which must be gray).
+              - the content area is opaque (it must be gray).
+            when unchecked:
+              - the area behind tabs is transparent (it must be green).
+              - the tabs area is opaque (it must be red, except the selected tab which must be gray).
+              - the content area is opaque (it must be gray).
+
+            Check this behaviour by clicking on present L&F button and tab layout.
+
+            Test Case 2 - Test Content pane opacity:
+            To test Content pane opacity, make sure "Opaque checkbox" is UNCHECKED.
+
+            Verify the following with 'content opaque' option:
+            when checked:
+              - the content area should be opaque (it must be gray).
+            when unchecked:
+              - the content area should be transparent (it must be green).
+
+            Check this behaviour by clicking on present L&F button and tab layout.
+
+            Test Case 3 - Test Tabs opacity:
+            To test Tabs opacity, make sure "Opaque checkbox" is UNCHECKED.
+            NOTE: For Nimbus LAF, tabs color are specific to nimbus style.
+                  All tabs are gray in color if tabs opaque is checked.
+
+            Verify the following with 'tabs opaque' option:
+            when checked:
+              - the tabs are opaque (it must be red, except the selected tab which must be gray).
+            when unchecked:
+              - the tabs are transparent (it must be green).
+
+            Check this behaviour by clicking on present L&F button and tab layout.""";
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+            .title("JTabbedPane Tab and Content Area Color Test Instructions")
+            .instructions(INSTRUCTIONS)
+            .testTimeOut(10)
+            .rows(25)
+            .columns(60)
+            .testUI(TestJTabbedPaneOpaqueColor::createAndShowUI)
+            .build()
+            .awaitAndCheck();
+    }
+
+    private static JFrame createAndShowUI() {
+        int NUM_TABS = 15;
+        frame = new JFrame("Test JTabbedPane Opaque Color");
+        JTabbedPane tabPane = new JTabbedPane();
+        tabPane.setTabLayoutPolicy(JTabbedPane.SCROLL_TAB_LAYOUT);
+        tabPane.setTabPlacement(JTabbedPane.TOP);
+        PassFailJFrame.addTestWindow(frame);
```

Comment:
Is it required to add `frame` explicitly to PassFailJFrame here?

Reply:
Looks redundant to me, removed.